packages/gestalt-datepicker/src/DatePicker.js

```diff
@@ -113,15 +113,15 @@ export type Props = {|
   /**
    * Required for date range selection. End date on a date range selection. See the [date range example](https://gestalt.pinterest.systems/web/datepicker#Date-range) to learn more.
    */
-  rangeEndDate?: Date,
+  rangeEndDate?: Date | null,
```

Comment:
Does the question mark take care of this?

Reply:
Or does it explicitly have to be null?

Reply:
rangeEndDate might not always be used, it's just for ranges. But when used the value might come as null when changed. This value is dynamic, different from minDate or maxDate

Reply:
(I'm assuming you're referring to the question mark after `rangeEndDate`)

Question marks in Flow generally convey "optional", but there are subtle differences when that is applied to the prop name vs the prop value.

```js
{| propName?: 'a' | 'b' |}
```
`propName` may be supplied, or it may be omitted. If it's supplied, it _must_ be either `'a'` or `'b'`. `propName={undefined}` is an error.

```js
{| propName: ?boolean |}
```
`propName` is a required prop — omitting it is an error. When `?` is applied to a prop value, it translates to `propValue | null | void` (`void` is `undefined`). So in this case `propName={undefined}` is fine, as would be `null` or a boolean.

```js
{| propName?: ?boolean |}
```
`propName` is an optional prop and can be omitted. If it's supplied, it can be `undefined`, `null`, or a boolean. This is the most flexible option but also requires the most type checking within the component logic.

I've found that generally I like optional props with strongly defined types for components — the prop can be omitted, but if it's present, we have a good idea of its type. In places like this one where we do need some sort of nil type as well, I prefer the approach Alberto is taking here: `null` never happens by accident (unlike `undefined`) so it conveys intention, and specifically using `Date | null` is stronger than `?Date`, where `undefined` would also be acceptable. (This is similar to [my preferred approach for typing objects](https://github.com/pinterest/gestalt/pull/3005#discussion_r1224878077))